### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by TomaszLach2**

### CODE REVIEW

**Review by Codeweld — CHANGES_REQUESTED** [review]

File: tests/Integration/ChannelContext/ChannelContextTest.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    public function testThrowExceptionWhenChannelIsNull(): void
+    {
+        //Then
+        $this->expectException(ChannelNotFoundException::class);
+
+        //When
+        $this->channelContext->getChannel();
+    }
+
+    public function testThrowExceptionWhenChannelCodeIsNull(): void
```
[the remark this answers is not in the document]
Comment (TomaszLach2) [reply tests/Integration/ChannelContext/ChannelContextTest.php:61]: Unneceseary test removed.

File: tests/Integration/ChannelContext/ChannelContextTest.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        $this->channelContext->setChannelCode(ChannelFixture::CHANNEL_CODE);
+        $channelFromContext = $this->channelContext->getChannel();
+        $channelCodeFromContext = $this->channelContext->getChannelCode();
+
+        //Then
+        self::assertSame($channel, $channelFromContext);
+        self::assertSame($channel->getCode(), $channelCodeFromContext);
+    }
+
+    public function testThrowExceptionWhenChannelIsNull(): void
```
[the remark this answers is not in the document]
Comment (TomaszLach2) [reply tests/Integration/ChannelContext/ChannelContextTest.php:52]: Unneceseary test removed.

File: tests/Integration/Event/Admin/PushNotificationConfigurationEventSubscriberTest.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->pushNotificationConfigurationEventSubscriber = self::getContainer()->get(PushNotificationConfigurationEventSubscriber::class);
+        $this->pushNotificationConfigurationRepository = self::getContainer()->get(PushNotificationConfigurationRepository::class);
+        $this->channelRepository = self::getContainer()->get(ChannelRepository::class);
+    }
+
+    public function testUploadWithoutFile(): void
```
[the remark this answers is not in the document]
Comment (TomaszLach2) [reply tests/Integration/Event/Admin/PushNotificationConfigurationEventSubscriberTest.php:35]: Unneceseary test removed.

File: tests/Application/config/packages/doctrine.yaml [hunk 4]
```diff
@@ -12,3 +12,8 @@ doctrine:
         charset: UTF8
 
         url: '%env(resolve:DATABASE_URL)%'
+
+when@test:
+    doctrine:
+        dbal:
+            dbname_suffix: '_test%env(default::TEST_TOKEN)%'
```
[the remark this answers is not in the document]
Comment (TomaszLach2) [reply tests/Application/config/packages/doctrine.yaml:19]: Yes, I am working on a test database.
https://symfony.com/doc/current/the-fast-track/en/17-tests.html#working-with-a-test-database

File: tests/Integration/Event/Admin/PushNotificationConfigurationEventSubscriberTest.php [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        $channel = $this->channelRepository->findOneBy(['code' => ChannelFixture::CHANNEL_CODE]);
+        /** @var PushNotificationConfiguration $configuration */
+        $configuration = $this->pushNotificationConfigurationRepository->findOneBy(['channel' => $channel]);
+
+        //When
+        $this->pushNotificationConfigurationEventSubscriber->prePushNotificationConfigurationUpdate(
+            new ResourceControllerEvent($configuration),
+        );
+
+        self::assertNull($configuration->getIconPath());
```
[the remark this answers is not in the document]
Comment (TomaszLach2) [reply tests/Integration/Event/Admin/PushNotificationConfigurationEventSubscriberTest.php:56]: Unneceseary test removed.

File: tests/Application/config/services_test.yaml [hunk 6]
```diff
@@ -1,8 +1,14 @@
 imports:
-    - { resource: "../../Behat/Resources/services.xml" }
     - { resource: "../../../vendor/sylius/sylius/src/Sylius/Behat/Resources/config/services.xml" }
 
 # workaround needed for strange "test.client.history" problem
 # see https://github.com/FriendsOfBehat/SymfonyExtension/issues/88
 services:
     Symfony\Component\BrowserKit\AbstractBrowser: '@test.client'
+
+    SpearDevs\SyliusPushNotificationsPlugin\:
```
[the remark this answers is not in the document]
Comment (TomaszLach2) [reply tests/Application/config/services_test.yaml:9]: I understand, this declaration is unnecessary.

**Review by Codeweld — APPROVED** [review]

**Pull request merged**